### PR TITLE
Updating multiple tasks with button 'Update'

### PR DESCRIPTION
**Pull request by agustin-pomies**
Tasks now can be updated to the database by the signed user in the web page.

### CODE REVIEW

File: app/assets/javascripts/tasks.coffee [hunk 1]
```diff
@@ -1,3 +1,27 @@
...
+      dataType: "script"
+      data:
+        task:
+          completed: $("input[name='task[completed]']").is(':checked')
+    alert "Hola Pomi, funciono!!"
+    #$().toggle()
+
+  #
+  # if $(".check")[0].checked == false
+  #  alert "Hola Pomi, funciono!!"
```
Comment (guillermoap) [inline app/assets/javascripts/tasks.coffee]: remove this and other commented code

File: app/controllers/tasks_controller.rb [hunk 2]
```diff
@@ -26,6 +29,20 @@ def create
...
 
+  def update
+    @task = Task.find(params[:id])
+    @task.update_attributes(task_params)
+  end
+
+  def update_multiple
+    @selected = params[:tasks_ids]
+
+    Task.all.update_all(completed: false) # completed_date will be overwrite when the task becomes completed
```
Comment (guillermoap) [inline app/controllers/tasks_controller.rb]: remove the `.all`

File: app/assets/javascripts/tasks.coffee [hunk 3]
```diff
@@ -1,3 +1,27 @@
...
+  # Update completed field for task when it's clicked
+  $(".check").change ->
+    $.ajax
+      url: task_path(@task)
+      type: "PUT"
+      dataType: "script"
+      data:
+        task:
+          completed: $("input[name='task[completed]']").is(':checked')
+    alert "Hola Pomi, funciono!!"
```
Comment (guillermoap) [inline app/assets/javascripts/tasks.coffee]: remove alerts like this

File: app/controllers/tasks_controller.rb [hunk 4]
```diff
@@ -26,6 +29,20 @@ def create
     end
   end
 
+  def update
+    @task = Task.find(params[:id])
```
Comment (guillermoap) [inline app/controllers/tasks_controller.rb]: add a simple error handling, remember that update return true\false depending of the success\failure of the update

File: app/controllers/tasks_controller.rb [hunk 5]
```diff
@@ -26,6 +29,20 @@ def create
...
+
+    Task.all.update_all(completed: false) # completed_date will be overwrite when the task becomes completed
+    Task.where(:id => @selected, :completed => false).update_all(completed: true, completed_date: Time.now)
+
+    redirect_to '/'
+  end
+
   def destroy
     @task = Task.find(params[:id])
     @task.destroy
```
Comment (guillermoap) [inline app/controllers/tasks_controller.rb]: same comment about the error handling as before

File: Gemfile [hunk 6]
```diff
@@ -63,3 +63,6 @@ gem 'tzinfo-data', platforms: [:mingw, :mswin, :x64_mingw, :jruby]
 
 # Gem to manipulate user authentication
 gem 'devise'
+
+# Gem to use jquery on rails
+gem 'jquery-rails'
```
Comment (santiago-bertinat) [inline Gemfile]: add version

File: app/assets/javascripts/tasks.coffee [hunk 7]
```diff
@@ -1,3 +1,27 @@
 # Place all the behaviors and hooks related to the matching controller here.
 # All this logic will automatically be available in application.js.
 # You can use CoffeeScript in this file: http://coffeescript.org/
+
+# $(document).on "ajax:success", "form#comments-form", (ev,data)->
+#  $("#comments-box").append("<li> #{data.body} - #{} </li>")
+
+$ ->
```
Comment (santiago-bertinat) [inline app/assets/javascripts/tasks.coffee]: lets not use coffee please

File: app/assets/javascripts/tasks.coffee [hunk 8]
```diff
@@ -1,3 +1,27 @@
...
+  $(document).on "change", "input[type~=checkbox]", ->
+  url = $(this).data("url")
+  $.ajax(url, dataType: "script", method: "POST")
+
+  # Update completed field for task when it's clicked
+  $(".check").change ->
```
Comment (santiago-bertinat) [inline app/assets/javascripts/tasks.coffee]: aren't you updating the tasks all at once? I think this is not necessary

File: app/assets/javascripts/tasks.coffee (hunk 1 above)
Comment (agustin-pomies) [reply app/assets/javascripts/tasks.coffee]: Resuelto en nuevo pull request

File: app/controllers/tasks_controller.rb (hunk 2 above)
Comment (agustin-pomies) [reply app/controllers/tasks_controller.rb]: Resuelto en nuevo pull request

File: app/assets/javascripts/tasks.coffee (hunk 3 above)
Comment (agustin-pomies) [reply app/assets/javascripts/tasks.coffee]: Resuelto en nuevo pull request

File: app/assets/javascripts/tasks.coffee (hunk 8 above)
Comment (agustin-pomies) [reply app/assets/javascripts/tasks.coffee]: Resolved in new pull request

File: app/assets/javascripts/tasks.coffee (hunk 7 above)
Comment (agustin-pomies) [reply app/assets/javascripts/tasks.coffee]: I will take it into account, thanks for the suggestion

File: Gemfile (hunk 6 above)
Comment (agustin-pomies) [reply Gemfile]: Okay, I will add this in my next commit

File: app/controllers/tasks_controller.rb (hunk 4 above)
Comment (agustin-pomies) [reply app/controllers/tasks_controller.rb]: Thanks for your suggestion. I will add this change in my next commit.

File: app/controllers/tasks_controller.rb (hunk 5 above)
Comment (agustin-pomies) [reply app/controllers/tasks_controller.rb]: Thanks for your suggestion. I will add this change in my next commit.

**Pull request merged**